src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.test.js

```diff
@@ -533,14 +401,72 @@ describe('convertInclude', () => {
       platform: 'highweb',
       type: 'include',
     };
-    const output = await convertInclude(input);
+    const output = await convertInclude(input, null, null, canonicalPathname);
     expect(fetch).toHaveBeenCalledTimes(1);
-    expect(output.model.html).toEqual(null);
-    expect(loggerMock.info).not.toHaveBeenCalled();
+    expect(output.model.html).toEqual(undefined);
+    expect(loggerMock.info).toHaveBeenCalledTimes(1);
     expect(loggerMock.error).toHaveBeenCalledTimes(1);
     expect(loggerMock.error).toHaveBeenCalledWith(INCLUDE_ERROR, {
       error: 'Error: this is an error message',
       url: 'https://foobar.com/includes/indepthtoolkit/quizzes/123-456',
     });
   });
+
+  it('should return ampSrc if AMP is supported and include is a VJ on AMP', async () => {
+    fetch.mockResponse(() => Promise.resolve('No fetch call'));
+    const includeSupportingAmp =
+      '/include/newsspec/21841-green-diet/gahuza/app?responsive=true&newsapps=true&app-image=https://news.files.bbci.co.uk/vj/live/idt-images/image-slider-asdf/app_launcher_ws_640_7ania.png&app-clickable=true&amp-clickable=true&amp-image-height=360&amp-image-width=640&amp-image=https://news.files.bbci.co.uk/vj/live/idt-images/image-slider-asdf/app_launcher_ws_640_7ania.png';
+    const input = {
+      required: false,
+      tile: 'Include from VisJo',
+      href: includeSupportingAmp,
+      platform: 'highweb',
+      type: 'include',
+    };
+    const expected = {
+      type: 'include',
+      model: {
+        href: includeSupportingAmp,
+        type: 'vj',
+        ampSrc:
+          'https://foobar.com/includes/amp/include/newsspec/21841-green-diet/gahuza/app/amp?responsive=true&newsapps=true&app-image=https://news.files.bbci.co.uk/vj/live/idt-images/image-slider-asdf/app_launcher_ws_640_7ania.png&app-clickable=true&amp-clickable=true&amp-image-height=360&amp-image-width=640&amp-image=https://news.files.bbci.co.uk/vj/live/idt-images/image-slider-asdf/app_launcher_ws_640_7ania.png',
```

Comment:
`https://foobar.com/includes/amp/include`, we shouldn't have `/amp` inserted between `/includes' and `/include, I think we have a bug somewhere, probably in `ampSrcBuilder`. Maybe make a similar test case to this for `ampSrcBuilder` and see what you can find?

Reply:
I think I just made a mistake in this test file setting the `process.env.SIMORGH_INCLUDES_BASE_AMP_URL` incorrectly rather than it being a bug in `ampSrcBuilder`?
https://github.com/bbc/simorgh/pull/6873/files?file-filters%5B%5D=.js#diff-754f58c9944064fefc2f77cd295825f9R26

Reply:
I don't think it's a bug. The href begins with a `/include` and the mock for process.env.SIMORGH_INCLUDES_BASE_AMP_URL is `https://foobar.com/includes/amp/include` which combines to result in the weird format, we could maybe remove the `/include` in `includeSupportingAmp`

```
const includeSupportingAmp =
  '/include/newsspec/...
```

Reply:
Yeah doesn't look like a bug, maybe just change the base url to differentiate amp from non-amp like this: https://github.com/bbc/simorgh/pull/6873/files#r442018884 

In any case looks like it's just the test 😄